tsfresh/feature_selection/selection.py

```diff
@@ -124,7 +136,141 @@ def select_features(X, y, test_for_binary_target_binary_feature=defaults.TEST_FO
+
+    :param n_jobs: Number of processes to use during the p-value calculation
+    :type n_jobs: int
+
+    :param chunksize: Size of the chunks submitted to the worker processes
+    :type chunksize: int
+
+    :return: A pandas.DataFrame with each column of the input DataFrame X as index with information on the significance
+             of this particular feature. The DataFrame has the columns
+             "Feature",
+             "type" (binary, real or const),
+             "p_value" (the significance of this feature as a p-value, lower means more significant)
+             "relevant" (True if the Benjamini Hochberg procedure rejected the null hypothesis for this feature)
+    :rtype: pandas.DataFrame
+    """
+    if ml_task not in ['auto', 'classification', 'regression']:
+        raise ValueError('ml_task must be one of: \'auto\', \'classification\', \'regression\'')
+    elif ml_task == 'auto':
+        ml_task = infer_ml_task(y)
+
+    if ml_task == 'classification':
+        relevance_tables = []
+        for label in y.unique():
+            y_label = (y == label)
+            relevance_table = check_fs_sig_bh(
+                X, y_label, target_is_binary=True, n_jobs=n_jobs, chunksize=chunksize,
+                test_for_binary_target_real_feature=test_for_binary_target_real_feature,
+                fdr_level=fdr_level, hypotheses_independent=hypotheses_independent,
+            )
+            relevance_tables.append((label, relevance_table))
+        relevance_table = combine_relevance_tables(relevance_tables)
+    elif ml_task == 'regression':
+        relevance_table = check_fs_sig_bh(
+            X, y, target_is_binary=False, n_jobs=n_jobs, chunksize=chunksize,
+            test_for_binary_target_real_feature=test_for_binary_target_real_feature,
+            fdr_level=fdr_level, hypotheses_independent=hypotheses_independent,
+        )
+
+    return relevance_table
+
+
+def combine_relevance_tables(relevance_tables_with_label):
+    """
+    Create a combined relevance table out of a list of tuples consisting of a target label
+    and its corresponding relevance table.
+    The combined relevance table containing the p_values for all target labels.
+
+    :param relevance_tables_with_label: A list of tuples: label, relevance table
+    :type relevance_tables_with_label: List[Tuple[Any, pd.DataFrame]]
+    :return: The combined relevance table
+    :rtype: pandas.DataFrame
+    """
+    def _append_label_to_p_value_column(a):
+        label, df = a
+        return df.rename(columns={'p_value': 'p_value_{}'.format(label)})
+
+    def _combine(a, b):
+        a.relevant |= b.relevant
+        return a.join(b.iloc[:,3])
 
-    return X.loc[:, df_bh[df_bh.rejected].Feature]
+    relevance_tables = map(_append_label_to_p_value_column, relevance_tables_with_label)
+    relevance_table = reduce(_combine, relevance_tables)
+    relevance_table['p_value'] = relevance_table.iloc[:, 3:].values.min(axis=1)
```

Comment:
> But I thought about that too, we could run the Benjamini-Hochberg-Test on the minimal p-values from all classes. That way, the FDR will be respected globally instead of just per class.

Mathematically, I do not that that is the right way. I think, what you have to do is to combine all p_values and then do BH test. So if you have three classes, you combine 3*p (p=number of features) p-Values to one BH test. You keep the features, if at least one of its p-values got rejected